src/backend/cdb/cdbmetadatacache_process.c

```diff
@@ -443,7 +443,7 @@ ProcessMetadataCacheCheck()
             DestroyHdfsFileInfo(file_info);
             total_remove_files++;
 
-            double cache_entry_ratio = hash_get_num_entries(MetadataCache) / metadata_cache_max_hdfs_file_num;
+            double cache_entry_ratio = (hash_get_num_entries(MetadataCache) * 1.0) / metadata_cache_max_hdfs_file_num;
```

Comment:
It would be better to use:
double cache_entry_ratio = ((double) hash_get_num_entries(MetadataCache)) / metadata_cache_max_hdfs_file_num;